src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js

```diff
@@ -16,6 +41,7 @@ function AttachmentViewPdf({file, encryptedSourceUrl, isFocused, isUsedInCarouse
 
     const onScaleChanged = useCallback(
         (scale) => {
+            updateScale(scale);
```

Comment:
Looking again at these props, I think we can just use `onScaleChanged` to pass the scale down, and then override it in the

```diff
 function AttachmentViewPdf(props) {
+    const {onScaleChanged, ...restProps} = props;
     const attachmentCarouselPagerContext = useContext(AttachmentCarouselPagerContext);
     const scaleRef = useSharedValue(1);
     const offsetX = useSharedValue(0);
@@ -49,8 +50,11 @@ function AttachmentViewPdf(props) {
                 >
                     <BaseAttachmentViewPdf
                         // eslint-disable-next-line react/jsx-props-no-spreading
-                        {...props}
-                        updateScale={updateScale}
+                        {...restProps}
+                        onScaleChanged={(scale) => {
+                            updateScale(scale)
+                            onScaleChanged()
+                        }}
                     />
                 </Animated.View>
             </GestureDetector>
```

```diff
-            onScaleChangedProp();
+            onScaleChangedProp(scale);
```